core/Plugin/Visualization.php

```diff
@@ -207,9 +207,29 @@ protected function buildView()
             $view->error = $loadingError;
         }
 
+        $templateFile = static::TEMPLATE_FILE;
+        $templateVars = $this->templateVars;
+
+        /**
+         * Can be trigerred in order to extend or modify Report visualization output.
+         *
+         * For example, if you want to display additional info which is in the dataTable but was omitted in original
+         * report visualization.
+         *
+         * Value of $templateFile should keep to Piwik format '@<Plugin_Name>/<template_name>.twig'
+         *
+         * The values under templateVars are accesible directly in template for example value under
+         * $templateVars['example'] in template is available under {{ example }}
+         *
+         * @param array &$templateFile The original template file, the overriding one should extend the original one
+         * @param array &$templateFile Template variables, you can add new or modify the ones already present
+         */
+        Piwik::postEvent('Visualization.customizeTemplate', array(&$templateFile, &$templateVars));
```

Comment:
The problem is, it would be come API and we wouldn't be able to change it anymore.

Also a problem is that we do not really have a way yet, to document existing template variables etc